### PR TITLE
Improve MOOSE documentation

### PR DESCRIPTION
**Pull request by milljm**
Improve upon the markdown syntax for specialized functions.
Add memory_logger documentation.
Use Github's LFS for storing images.
Refs #6699 
Refs #7617

### CODE REVIEW

**Comment by milljm** [comment]
Referencing #7628, because it is relevant to this PR.

File: docs/content/moose_tools/memory_logger.md [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+
+## Basic usage
+In its simplest use case scenario, you encapsulate the command you would normally run with out Memory Logger as an argument to Memory Logger:
+```text
+memory_logger.py --run "mpiexec -n 2 /absolute/path/to/moose_test-opt -i simple_diffusion.i -r 5"
+```
+!!! Important
+    Always use absolute paths between those --run double quotes. Memory Logger has no knowledge on paths like: '../../  or ~/some/path'
+
+Memory Logger will launch your process as normal, and begin to track the two processes that launched. When Memory Logger detects that the process/s has exited (for good or for ill), it will explain where the log was saved. You can then instruct Memory Logger to open this file, and display its information either using Gnuplot (--plot), or by formatting the information and outputting it to stdout (--read).
```
Comment (aeslaughter) [inline docs/content/moose_tools/memory_logger.md]: You use matplotlib not Gnuplot, correct?

File: docs/content/moose_tools/memory_logger.md [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+12 May 09:43:46.615820        240,656 K | ---------------------------------------------------| 91.06%
+12 May 09:43:46.640438        244,368 K | ----------------------------------------------------| 92.47%
+12 May 09:43:46.665082        235,456 K | --------------------------------------------------| 89.10%
+12 May 09:43:46.688966        230,344 K | -------------------------------------------------| 87.16%
+12 May 09:43:46.713106        195,528 K | ----------------------------------------| 73.99%
+12 May 09:43:46.739572        177,140 K | -----------------------------------| 67.03%
+12 May 09:43:46.767171              0 K | | 0%
+Date Stamp                 Memory Usage | Percent of MAX memory used: ( 264,256 K )
+```
+Okay... A bit too accurate, as several samples remained unchanged during tracking. However sometimes more is desirable, though just not printed in this fashion. For data dumps like this, using Gnuplot is far more efficient.
```
Comment (aeslaughter) [inline docs/content/moose_tools/memory_logger.md]: matplotlib?

File: docs/content/moose_tools/memory_logger.md [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+
+## Using Gnuplot
```
Comment (aeslaughter) [inline docs/content/moose_tools/memory_logger.md]: matplotlib

File: docs/content/moose_tools/memory_logger.md [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+<img width="300" src="../../../media/memory_logger-plot_multi.png">
+
+We can visualize the results by plotting the data with Gnuplot:
```
Comment (aeslaughter) [inline docs/content/moose_tools/memory_logger.md]: matplotlib

File: docs/content/moose_tools/memory_logger.md [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+Obtaining stack traces while tracking memory usage on a single machine or across PBS nodes is the key feature of this tool. In order to do so we need to supply an additional two arguments; `--pstack` and `--debugger gdb|lldb`. Memory Logger supports two debuggers for the purpose of obtaining a stack trace (gdb or lldb).
+```text
+memory_logger.py --pstack --debugger gdb \
+--run "mpiexec /absolute/path/to/moose_test-opt -i simple_diffusion.i -r 5"
+```
+The --pstack argument is used for both tracking and plotting. It tells Memory Logger to actually display stack trace information (if available in the log file).
+```text
+memory_logger.py --pstack --plot simple_diffusion_memory.log
+```
+Stack traces are represented in the form of points along the Gnuplot line graph. Clicking on these dots will cause memory_logger.py to print the actual stack trace gleamed at that time to stdout. However, if you choose to display all this junk on the terminal, you can with --read:
```
Comment (aeslaughter) [inline docs/content/moose_tools/memory_logger.md]: matplotlib

File: docs/content/moose_tools/memory_logger.md [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+```text
+memory_logger.py --pstack --read simple_diffusion_memory.log
+<not going to paste this data here. I am telling ya, its pages and pages>
+```
+You can also display stdout along the Gnuplot graph:
```
Comment (aeslaughter) [inline docs/content/moose_tools/memory_logger.md]: matplotlib

File: docs/content/moose_tools/memory_logger.md (hunk 1 above)
Comment (aeslaughter) [inline docs/content/moose_tools/memory_logger.md]: You can probably change this to say 'matplotlib' and then just refer to "plot" or "graph" at the other locations.

**Pull request merged**